### PR TITLE
 New features from old technical task

### PR DESCRIPTION
**Pull request by IgorBMSTU**
- added copy_field query,
- eaf_search query that uses poio-api library
- added new metadata fields
- bulk_update_entity content

### CODE REVIEW

File: lingvodoc/schema/query.py [hunk 1]
```diff
@@ -502,6 +506,16 @@ class Query(graphene.ObjectType):
         graphene.List(graphene.String), id=graphene.Int(required=False))
     perspectives_fields_intersection = graphene.Field(
         graphene.List(Field), perspectives=graphene.List(LingvodocID),)
+    eaf_search = graphene.Field(graphene.String, pid=LingvodocID(required=True), searchstring=graphene.Argument(ObjectVal))
+
+    def resolve_eaf_search(self, info, pid, searchstring):
+        storage = dict()
+        storage['path'] = "/home/andriy/Desktop/objects/"
```
Comment (IgorBMSTU) [inline lingvodoc/schema/query.py:513]: todo: reading from config (ex: https://github.com/ispras/lingvodoc/blob/75bd898f9b3c4fa447f079a66de084435aac9f97/alembic/env.py#L39)

File: lingvodoc/utils/search.py [hunk 2]
```diff
@@ -108,6 +115,69 @@ def eaf_words(eaf_obj):
...
+        extension = entity.content[entity.content.rfind('.'):]
+        url = ascii_part + urllib.request.quote(unicode_part) + extension
+
+        response = urllib.request.urlopen(url)
+        tmp_content_file_path = ''.join([
+            storage_path, '/', 'tmp_content_file'])
+        tmp_content_file = open(tmp_content_file_path, "wb+")
+        tmp_content_file.write(response.read())
+
+        eaf_search(tmp_content_file_path, xls_path, searchstring,
```
Comment (IgorBMSTU) [inline lingvodoc/utils/search.py:171]: TODO: use multiprocess

File: server-requirements.txt [hunk 3]
```diff
@@ -47,7 +47,14 @@ pathvalidate==0.8.3
...
 promise==2.1
 iso8601==0.1.11
+wheel
+numpydoc
+regex
+graf-python
+tablib==0.13.0
+xlwt==1.3.0
+xlutils
+git+https://github.com/ankan2013/poio-api.git
```
Comment (IgorBMSTU) [inline server-requirements.txt]: TODO: using a reliable repository

**Comment by IgorBMSTU** [comment]
query test{
  eaf_search(pid:[2400,11],searchstring:
{type: "or", value:
[
    {
      type: "or", value: 
      [
        {type: "substr", value: "Ангел", tiers: "text"},
       {type: "str", value: "Ангелъ", tiers: "transcription"}
      ]
    }
},
{type: "substr", value: "a"}
]
  )
}


tiers необязательное поле

File: lingvodoc/schema/gql_copy_field.py [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+                                         save_object=True)
+                    dbentities_list.append(dbentity)
+
+        if len(dbentities_list) > 0:
+            DBSession.bulk_save_objects(dbentities_list)
+            DBSession.flush()
+
+        return CopySingleField(triumph=True)
+
+class CopySoundMarkupFields(graphene.Mutation):
```
Comment (BenderTrump) [inline lingvodoc/schema/gql_copy_field.py]: add multiprocessing

File: lingvodoc/utils/search.py [hunk 5]
```diff
@@ -108,6 +115,69 @@ def eaf_words(eaf_obj):
     annotations = set(annotations)
     return annotations
 
+
+def search_eaf_blocks_in_perspective(pid, storage, searchstring, client_id):
+    e1 = aliased(dbEntity)
+    le = aliased(dbLexicalEntry)
+    f = aliased(dbField)
```
Comment (BenderTrump) [inline lingvodoc/utils/search.py:122]: Need to check read permission

File: lingvodoc/utils/search.py [hunk 6]
```diff
@@ -108,6 +115,69 @@ def eaf_words(eaf_obj):
...
+                storage_path, '/', xls_name])
+
+    result_url = ''.join([storage['prefix'], storage['static_route'], folder, '/', xls_name])
+
+    open(xls_path, "w+").close()
+
+    for row in query_result:
+        entity = row[0]
+        field = row[2]
+        if field.data_type != 'Markup' or entity.content.find(".eaf") == -1:
```
Comment (BenderTrump) [inline lingvodoc/utils/search.py:157]: Better to filter field data type in db request

**Pull request merged**